ksql-engine/src/main/java/io/confluent/ksql/function/udf/math/Round.java

```diff
@@ -0,0 +1,139 @@
+ROUND(1.9) -> 2
+
+ROUND(-1.1) -> -1
+ROUND(-1.5) -> -1    Note this is not -2! We round up and up is in a positive direction.
+ROUND(-1.9) -> -2
+ */
+@UdfDescription(name = "Round", description = Round.DESCRIPTION)
+public class Round {
+
+  static final String DESCRIPTION =
+      "Returns the value rounded to the specified number of decimal places. "
+      + "If the number of decimal places is omitted it defaults to zero."
+      + "Numbers are rounded to the nearest digit at the last required decimal place. "
+      + "If the digit to the right of the last required decimal place is equidistant (i.e. a 5) "
+      + "then it is rounded up.";
+
+  private static final MathContext SIMPLE_ROUND_HALF_UP =
+      new MathContext(1, RoundingMode.HALF_UP);
+
+  @Udf
+  public Long round(@UdfParameter final Double val) {
+    return val == null ? null : Math.round(val);
+  }
+
+  @Udf
+  public Double round(@UdfParameter final Double val, @UdfParameter final Integer decimalPlaces) {
+    // Note that we MUST BigDecimal.valueOf(double) to create the BD to ensure correctness
+    // as this methods initialises via the string representation
+    return val == null
+        ?
+        null : roundBigDecimal(BigDecimal.valueOf(val), decimalPlaces).doubleValue();
+  }
+
+  @Udf(schemaProvider = "provideDecimalSchema")
+  public BigDecimal round(@UdfParameter final BigDecimal val) {
+    return val == null ? null : round(val, 0);
+  }
+
+  @Udf(schemaProvider = "provideDecimalSchemaWithDecimalPlaces")
+  public BigDecimal round(
+      @UdfParameter final BigDecimal val,
+      @UdfParameter
+      final Integer decimalPlaces
+  ) {
+    return val == null ? null : roundBigDecimal(val, decimalPlaces);
+  }
+
+  @UdfSchemaProvider
+  public SqlType provideDecimalSchemaWithDecimalPlaces(final List<SqlType> params) {
+    if (params.size() != 2) {
+      throw new KsqlException("Round udf with decimal places accepts two parameters");
+    }
+    final SqlType s0 = params.get(0);
+    if (s0.baseType() != SqlBaseType.DECIMAL) {
+      throw new KsqlException("The schema provider method for round expects a BigDecimal parameter"
+          + "type as first parameter.");
+    }
+    final SqlType s1 = params.get(1);
+    if (s1.baseType() != SqlBaseType.INTEGER) {
+      throw new KsqlException("The schema provider method for round expects an Integer parameter"
+          + "type as second parameter.");
+    }
+    return s0;
+  }
+
+  @UdfSchemaProvider
+  public SqlType provideDecimalSchema(final List<SqlType> params) {
+    if (params.size() != 1) {
+      throw new KsqlException("Round udf accepts one parameter");
+    }
+    final SqlType s0 = params.get(0);
+    if (s0.baseType() != SqlBaseType.DECIMAL) {
+      throw new KsqlException("The schema provider method for round expects a BigDecimal parameter"
+          + "type as a parameter.");
+    }
+    return s0;
+  }
+
+  /*
+  Unfortunately there is an inconsistency in the way that java.lang.Math and
```

Comment:
I would move this kind of comment from the code and into this PR and/or the commit message of the final squash merge.  

This removes the verbosity from the code and means anyone that wants to understand the reasons why the code is the way it is can check the PR/commit.  

Also, as before, what's really the most important is to have tests that fail should someone switch the implementation to something incompatible.

Reply:
I'm a believer in having minimal comments but I don't believe in zero comments ;) 
I think this comment is justified because without the reader of the code probably won't understand why it is done this way, and might not think of going to the commit to check. Also going to the commit is a PITA compared to reading the comment.

Reply:
consider moving to proper java docs on the class if you _really_ want to keep all of these comments.

Reply:
I'll move it to the header

Reply:
Done